lib/core/amount_validator.dart

```diff
@@ -1,32 +1,68 @@
 import 'package:cake_wallet/core/validator.dart';
 import 'package:cake_wallet/generated/i18n.dart';
-import 'package:cw_core/wallet_type.dart';
+import 'package:cw_core/crypto_currency.dart';
 
 class AmountValidator extends TextValidator {
-  AmountValidator({required WalletType type, bool isAutovalidate = false})
+  AmountValidator({required CryptoCurrency currency, bool isAutovalidate = false}) {
+    symbolsAmountValidator =
+        SymbolsAmountValidator(currency: currency, isAutovalidate: isAutovalidate);
+    decimalAmountValidator = DecimalAmountValidator(currency: currency);
+  }
+
+  late final SymbolsAmountValidator symbolsAmountValidator;
+
+  late final DecimalAmountValidator decimalAmountValidator;
+
+  String? call(String? value) => symbolsAmountValidator(value) ?? decimalAmountValidator(value);
+}
+
+class SymbolsAmountValidator extends TextValidator {
+  SymbolsAmountValidator({required CryptoCurrency currency, required bool isAutovalidate})
       : super(
             errorMessage: S.current.error_text_amount,
-            pattern: _pattern(type),
+            pattern: _pattern(currency),
+            isAutovalidate: isAutovalidate,
+            minLength: 0,
+            maxLength: 0);
+
+  static String _pattern(CryptoCurrency currency) {
+    switch (currency) {
+      case CryptoCurrency.xmr:
+        return '^([0-9]+([.\,][0-9]+)?|[.\,][0-9]+)\$';
+      case CryptoCurrency.btc:
+        return '^([0-9]+([.\,][0-9]+)?|[.\,][0-9]+)\$';
+      default:
+        return '^([0-9]+([.\,][0-9]+)?|[.\,][0-9]+)\$';
+    }
+  }
+}
+
+class DecimalAmountValidator extends TextValidator {
+  DecimalAmountValidator({required CryptoCurrency currency, bool isAutovalidate = false})
+      : super(
+            errorMessage: S.current.decimal_places_error,
+            pattern: _pattern(currency),
             isAutovalidate: isAutovalidate,
             minLength: 0,
             maxLength: 0);
 
-  static String _pattern(WalletType type) {
-    switch (type) {
-      case WalletType.monero:
-        return '^([0-9]+([.\,][0-9]{0,12})?|[.\,][0-9]{1,12})\$';
-      case WalletType.bitcoin:
-        return '^([0-9]+([.\,][0-9]{0,8})?|[.\,][0-9]{1,8})\$';
+  static String _pattern(CryptoCurrency currency) {
+    switch (currency) {
+      case CryptoCurrency.xmr:
+        return '^([0-9]+([.\,][0-9]{1,12})?|[.\,][0-9]+)\$';
+      case CryptoCurrency.btc:
+        return '^([0-9]+([.\,][0-9]{1,8})?|[.\,][0-9]+)\$';
```

Comment:
please validate those RegEx and we can instead of adding all the currencies, just set the `default` case to the average amount validation, which I believe would be the same as the `btc` RegEx, but feel free to verify that too